boxcli/add.go

```diff
@@ -32,7 +32,13 @@ func AddCmd() *cobra.Command {
 
 func addCmdFunc(flags *addCmdFlags) runFunc {
 	return func(cmd *cobra.Command, args []string) error {
-		box, err := devbox.Open(flags.config.path, os.Stdout)
+		dir := flags.config.path
+		if dir == "" && devbox.IsDevboxShellEnabled() {
```

Comment:
I think I missed some of the discussion around this feature, so apologies if I'm rehashing some stuff.

I'm trying to think through what the expected behavior would be here.

- If I'm in a devbox shell and cd into another devbox directory, is it obvious that add/rm will be operating on the shell's config and not the current directory's config?
- Similarly, it's weird that add and remove will affect a different config than all of the other commands. Or are we going to leave all the other commands disabled?
- What happens if my devbox directory is renamed or moved while I'm in a shell?
- Does this feature provide a big enough benefit now that we search parent directories for the config?

I want to say that if we're adding this feature, then we should disallow setting the config flag when in a shell and print a warning when the env var is overriding another devbox.json. Also, if the path in the env var becomes invalid, then we should tell the user to restart their shell. Thoughts?

Reply:
oh ho 🤦🏾 I was on auto-pilot and rebased this PR onto the `--config` flag parent PR, without thinking it through.

With the new `--config` flag functionality, this PR is not needed! yay!

Abandoning.